tests/basic/pool_function_test.go

```diff
@@ -899,73 +899,77 @@ var _ = Describe("{PoolExpandAddDiskInMaintenanceMode}", func() {
 			3. Exit out of maintenance mode (PX only performs pool expand in normal mode, not in maintenance mode)
 			4. Verify pool expand operation goes to completion.
 	*/
-
-	BeforeEach(func() {
-		StartTorpedoTest("PoolExpandAddDiskInMaintenanceMode",
-			"Initiate pool expand with add-disk when node is already in maintenance mode", nil, testrailID)
-		contexts = scheduleApps()
-	})
-
-	JustBeforeEach(func() {
-		poolIDToResize = pickPoolToResize()
-		log.Infof("Picked pool %s to resize", poolIDToResize)
-		poolToResize = getStoragePool(poolIDToResize)
-	})
-
-	JustAfterEach(func() {
-		AfterEachTest(contexts)
-	})
-
-	AfterEach(func() {
-		appsValidateAndDestroy(contexts)
-		EndTorpedoTest()
-	})
-
-	stepLog := "Start pool expand with add-disk on node which is already in maintenance mode "
-	It(stepLog, func() {
-		log.InfoD(stepLog)
-		var nodeDetail *node.Node
-		var err error
-		stepLog = "Move node to maintenance mode"
-		Step(stepLog, func() {
-			log.InfoD(stepLog)
-			nodeDetail, err = GetNodeWithGivenPoolID(poolToResize.Uuid)
-			dash.VerifyFatal(err, nil, fmt.Sprintf("Failed to get Node Details using PoolUUID [%v]", poolToResize.Uuid))
-
-			log.InfoD("Bring Node to Maintenance Mode")
-			err = Inst().V.EnterMaintenance(*nodeDetail)
-			dash.VerifyFatal(err, nil, fmt.Sprintf("Failed to shift Node [%s] to Mainteinance Mode", nodeDetail.Name))
+	Support, err := IsPoolAddDiskSupported()
```

Comment:
you are calling this function outside it , so this will called while ginkgo scans for the tests and will lead to nil pointer exceptions, Please change the logic as discussed

Reply:
calling the function now inside